src/protocols/bdx/AsyncTransferFacilitator.cpp

```diff
@@ -170,21 +181,20 @@ void AsyncResponder::NotifyEventHandled(const TransferSession::OutputEventType e
     ChipLogDetail(BDX, "NotifyEventHandled : Event %s Error %" CHIP_ERROR_FORMAT,
                   TransferSession::OutputEvent::TypeToString(eventType), status.Format());
 
-    // If this is the end of the transfer (whether a clean end, or some sort of error condition), ensure that
-    // we destroy ourselves after processing any output events that might have been generated by
-    // the transfer session to handle end-of-transfer (e.g. in some error conditions it might want to send
-    // out a status report).
+    // If this is the end of the transfer (whether a clean end, or some sort of error condition), ensure
+    // that we destroy ourselves after unwinding the processing loop in the ProcessOutputEvents API.
+    // We can ignore the status for these messages since the state machine is in a bad, unrecoverable
+    // state and we should stop processing events and clean up.
     if (eventType == TransferSession::OutputEventType::kAckEOFReceived ||
+        eventType == TransferSession::OutputEventType::kStatusReceived ||
         eventType == TransferSession::OutputEventType::kInternalError ||
-        eventType == TransferSession::OutputEventType::kTransferTimeout ||
-        eventType == TransferSession::OutputEventType::kStatusReceived)
+        eventType == TransferSession::OutputEventType::kTransferTimeout)
     {
         mDestroySelfAfterProcessingEvents = true;
     }
-
     // If there was an error handling the output event, this should notify the transfer object to abort transfer so it can send a
     // status report across the exchange when we call ProcessOutputEvents below.
```

Comment:
This commet should move inside the body of the "else if"; would be more readable.

Reply:
done.